lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -228,6 +235,59 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Split $this->_bindedValues into those values that need to be sent using mysqli::send_long_data()
+     * and those that can be bound the usual way.
+     *
+     * @return array
+     */
+    private function separateLongData()
+    {
+        $streams = $otherData = [];
+
+        foreach ($this->_bindedValues as $parameter => $value) {
+            if ($this->types[$parameter - 1] !== static::$_paramTypeMap[ParameterType::LARGE_OBJECT]) {
+                $otherData[$parameter] = $value;
+                continue;
+            }
+
+            if (is_resource($value)) {
+                if (get_resource_type($value) !== 'stream') {
+                    throw new InvalidArgumentException('Resources passed with the LARGE_OBJECT parameter type must be stream resources.');
+                }
+                $streams[$parameter]   = $value;
+                $otherData[$parameter] = null;
+            } else {
+                $this->types[$parameter - 1] = static::$_paramTypeMap[ParameterType::STRING];
```

Comment:
Can we return `$types` as part of the return value instead of modifying the object state? Look at `SQLParserUtils::expandListParameters()` for example.